Snabble/UI/DynamicStack/DynamicStackWidget.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Widget.swift
+//  Snabble
+//
+//  Created by Andreas Osberghaus on 31.08.22.
+//
+
+import Foundation
+
+public protocol Widget: Codable {
+    var id: String { get }
+    var type: WidgetType { get }
+}
+
+public enum WidgetType: String, Codable {
+    case text
+    case image
+    case button
+    case information
+    case purchases
+}
+
+public struct WidgetText: Widget, Codable {
+    public let id: String
+    public let type: WidgetType = .text
+    public let text: String
+    public let textColorSource: String
+    public let textStyleSource: String
+
+    enum CodingKeys: String, CodingKey {
+        case id
+        case text
+        case textColorSource = "textColor"
+        case textStyleSource = "textStyle"
+    }
+}
+
+public struct WidgetImage: Widget, Codable, ImageSourcing {
+    public let id: String
+    public let type: WidgetType = .image
+    public let imageSource: String?
+
+    enum CodingKeys: String, CodingKey {
+        case id
+        case imageSource = "image"
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        id = try container.decode(String.self, forKey: .id)
+        imageSource = try container.decode(String.self, forKey: .imageSource)
```

Comment:
Das habe ich gemacht, da `imageSource` eigentlich nicht optional ist, aber `ImageSourcing` imageSource als optional deklariert. Bin mir unsicher ob wir es so machen sollen.

Reply:
In diesem Fall sollte es nicht optional sein.

Reply:
[eb0914b](https://github.com/snabble/iOS-SDK/pull/170/commits/eb0914b00524facc5934e43d12174f35ed708981)